File: app/src/main/java/com/wl/androidlearning/designpattern/flyweight/ShapeFactory.java
package com.wl.androidlearning.designpattern.flyweight;

import java.util.HashMap;

public class ShapeFactory {

    private static final HashMap<String,Shape> circleMap = new HashMap();

    public static Circle createCircle(Circle circle){

        Circle resultCircle=(Circle)circleMap.get(circle.getColor());
        if(resultCircle==null){
            resultCircle = circle;
            circleMap.put(circle.getColor(),circle);
        }
        return resultCircle;


    }

}

File: app/src/main/java/com/wl/androidlearning/designpattern/factory/ShapeFactory.kt
package com.wl.androidlearning.designpattern.factory

object ShapeFactory {

    fun createShape(shapeType:String):Shape?{
        when(shapeType){
            "circle"->return Circle()
            "rectangle"->return Rectangle()
            "square"->return Square()
            else->return null
        }

    }


}